Source/WebCore/Modules/mediasession/MediaSession.cpp

```diff
@@ -239,14 +239,26 @@ void MediaSession::setActionHandler(MediaSessionAction action, RefPtr<MediaSessi
 {
     if (handler) {
         ALWAYS_LOG(LOGIDENTIFIER, "adding ", action);
-        m_actionHandlers.set(action, handler);
+        {
+            Locker lock { m_actionHandlersLock };
+            m_actionHandlers.set(action, handler);
+        }
         auto platformCommand = platformCommandForMediaSessionAction(action);
         if (platformCommand != PlatformMediaSession::NoCommand)
             PlatformMediaSessionManager::sharedManager().addSupportedCommand(platformCommand);
     } else {
-        if (m_actionHandlers.contains(action)) {
+        bool containsAction;
```

Comment:
I think my proposal was better. The code was already unnecessarily inefficient by doing a double hash map look up and now is worse with double locking. Why didn’t you go with my proposal, was there something wrong with it?

Reply:
It's because I misread what you wrote above. I thought you were saying to use double locking but you were saying to use the return value of HashMap::remove

Reply:
I'll change it shortly.

Reply:
Ok I think I've tightened up the locks efficiently now. I'll keep this in mind in the future - keep an eye out for inefficient lookups and look for chances to keep a lock as tight as possible to the calls to the protected object.